### ISSUE TITLE
Seriously slow?

### ISSUE THREAD

**Issue by kokizzu** [zxh0/jvm.go#16]
I tried to execute this code

```
import java.util.*;
class prime {
        public static void main(String[] args) {
                ArrayList<Integer> res = new ArrayList<Integer>();
                int last = 3;
                res.add(last);
                while(true) {
                        last = last + 2;
                        boolean prime = true;
                        for(int v : res) {
                                if(v*v>last) break;
                                if(last%v == 0) {
                                        prime = false;
                                        break;
                                }
                        }
                        if(prime) {
                                res.add(last);
                                if(res.size()%100000 == 0) System.out.println(last);
                                if(last>9999999) break;
                        }
                }
        }
}
```
but it took too much time `578.62`s and just 2 lines of result given, for comparison openjdk gives complete output within `1.23`s

**Comment by zxh0** [comment]
Hi, thanks for your issue. This project is coded for fun and learning. Besides that, it is in the baby stage(two months old, mostly programmed by one guy). So, do not expect too much : ) I will try my best to finish this project and make it run faster.

**Comment by kokizzu** [comment]
Btw this project is cool :cool: i'll look forward for the completion :+1:

**Comment by phaikawl** [comment]
@zxh0 could there be a bug or something related to this? I have a feeling that no matter how slow, it shouldn't take that long. I don't know anything but yeah, my 2 cents.

**Comment by zxh0** [comment]
Yes, it's too slow :( I did some cpu profile and code optimization yesterday, but I feel it's not the time to do this kind of stuff. For now, this jvm is still a toy of a big boy. Thread, ClassLoader, Java Memory Model, File, Socket... a lot of things are still not working(correctly). I will try to implement all of them and then go back to this issue.

**Comment by wishstudio** [comment]
While peeking at the code I think it's already using a threaded interpreter, right?
Then to achieve good performance you must implement JITs, which I suspect will take years to be competitive... (look at LuaJIT)
I agree it's better making things working than making them performant for now.

**Comment by zxh0** [comment]
Actually, writing a JIT is beyond my ability :(

**Comment by andrewchambers** [comment]
Go can't easily support a JIT currently because Go's memory model clashes with most ways of doing jit. Creating a JIT itself is not particularly hard.

**Comment by bpilot** [comment]
What if we instead transpiled Java to Golang, and then could get native execution speed?

Kind of like GWT transpiles Java bytecode to JavaScript.

**Comment by bmhm** [comment]
@bpilot why use go then? GCC has gcj which compiles Java to native x64 elf binaries.

**Comment by xiangzhai** [comment]
> Actually, writing a JIT is beyond my ability :(

How about migrate [VMKit](https://github.com/llvm-mirror/vmkit) to latest LLVM?
